functions/models/src/types/user.ts

```
@@ -42,6 +45,7 @@ export class User extends UserRegistration {
   readonly dateOfEnrollment: Date
   readonly invitationCode: string
   readonly lastActiveDate: Date
+  readonly phoneNumbers: string[]
```

I think we might just want to make this a singular thing; I think we can generally assume that a user only has single phone number where they can receive text messages? If we have multiple phone numbers, we should at least store the type of phone number (e.g. landline or mobile) to see that we only send text messages to mobile numbers. Landlines might not respond well to test messages.

---

I was more or less lazy here, since we would need to check, if there is already a phone number, if we make this a single optional value, right?

Or asked differently: What would happen, if a user with a phone number calls checkPhoneNumberVerification and it finishes successfully?

- We overwrite the existing phone number and ignore that the user may now no longer receive notifications over the old phone number, which may be unexpected.
- We keep the old one and throw an error, when we try to override it, but we allow the starting of the verification.
- We check in both `startPhoneVerification` and `checkPhoneVerification`, if there already is a phone number and throw, if that is the case.

---

I think we will have to enforce that a user has to first start with `startPhoneVerification` and then calls `checkPhoneVerification`. Just calling it with an existing number that is already verified should result in an error and calling it with a new number that was not added in `startPhoneVerification` should also result in an error. So if the user calls `checkPhoneVerification` first with a new number we should throw an error and direct them to start the process in a proper way.

The mobile apps need to support this flow in the UI and need to ensure that any changes to the phone number trigger the same workflow as a new number. I think the current approach to temporarily store the new number in a separate object to support that two step process and only store the phone number in the user document is a solid approach for that.

---

Discussed this in more detail: We will now allow multiple phone numbers, but the client applications will ask to remove other existing phone numbers once a new one has successfully verified. This will allow us to potentially be more flexible in the future, while also suggesting that the "normal" flow would be to replace a phone number. 

A client application should still generally also allow to delete a phone number, even if it is not replaced.